File: speech_to_text/ios/Classes/SpeechToTextPlugin.m
#import "SpeechToTextPlugin.h"
#import <speech_to_text/speech_to_text-Swift.h>

@implementation SpeechToTextPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftSpeechToTextPlugin registerWithRegistrar:registrar];
}
@end

File: speech_to_text/macos/Classes/SpeechToTextPlugin.swift
import Cocoa
import FlutterMacOS

public class SpeechToTextPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "speech_to_text", binaryMessenger: registrar.messenger)
    let instance = SpeechToTextPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
